prometheus-net/Advanced/ThreadSafeDouble.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Threading;
+
+namespace Prometheus.Advanced
+{
+    public struct ThreadSafeDouble
+    {
+        private long _value;
```

Comment:
Interlocked.Read works on longs.
Bit converter will allow us to store a double to a long with ease